### ISSUE TITLE
iOS platform tests should be run on plugin CI

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#40407]
## Use case

It is valuable to confirm platform-specific code has not regressed.  There are already a few iOS tests in [example apps](https://github.com/flutter/plugins/tree/master/packages/in_app_purchase/example/ios/in_app_purchase_pluginTests) and it would be nice if these didn't need to be run manually.  This will be more valuable as the number of supported platforms increases.

## Proposal

Create scripts/other infrastructure to allow platform-specific tests to be written and run as part of plugin CI.

**Comment by mklim** [comment]
Currently it's possible to write tests in a roundabout way. For IAP @cyanglaz and I opened the example app's platform code in an IDE, then added unit tests "to the example app" that really tested the underlying plugin code the example app depended on. The IDE used the example app as to build and ran everything with no problem.

We never hooked this into CI. Admittedly it's also conceptually confusing to have to write tests for a plugin in the example app that uses it instead of in the plugin itself.

**Comment by mklim** [comment]
I looked into this to add them, and it turns out we've actually had Java tests hooked into CI for over a year now. :man_facepalming: They were first added in https://github.com/flutter/plugins/pull/521. They're currently running in CI now: https://cirrus-ci.com/task/4661309406183424?command=main#L1496.

The good news is that this means we can add Java tests for plugins now. I looked around and there didn't appear to be any iOS equivalent of this, just the Java tests.
